In [1]:
#Loading Data
import pandas as pd
import matplotlib.pyplot as plt
import warnings 
warnings.filterwarnings('ignore')

In [2]:
volume_df = pd.read_csv('../../../data/Old_Data/Music_Sales_Volume_Data.csv')

## Cleaning Music Sales Volume Data

In [3]:
volume_df.shape

(261, 16)

In [4]:
volume_df.head(20)

Year of Year Date Format Format (copy) Metric  Value (Actual)  \
0                2000     CD            CD  Units      942.500000   
1                2001     CD            CD  Units      881.900000   
2                2002     CD            CD  Units      803.300000   
3                2003     CD            CD  Units      746.000000   
4                2004     CD            CD  Units      767.000000   
5                2005     CD            CD  Units      705.400000   
6                2006     CD            CD  Units      619.700000   
7                2007     CD            CD  Units      499.700000   
8                2008     CD            CD  Units      368.400000   
9                2009     CD            CD  Units      296.600000   
10               2010     CD            CD  Units      253.000000   
11               2011     CD            CD  Units      240.800000   
12               2012     CD            CD  Units      198.164387   
13               2013     CD            CD  Units      173.793303   
14               2014     CD            CD  Units      138.702363   
15               2015     CD            CD  Units      117.144052   
16               2016     CD            CD  Units       97.577071   
17               2017     CD            CD  Units       87.682500   
18               2018     CD            CD  Units       51.954276   
19               2019     CD            CD  Units       46.506496   

    Adjusted for Inflation Flag  Year  Year Date % of Total Volume  \
0                           NaN  2000       2000             87.3%   
1                           NaN  2001       2001             91.1%   
2                           NaN  2002       2002             93.4%   
3                           NaN  2003       2003             93.4%   
4                           NaN  2004       2004             80.1%   
5                           NaN  2005       2005             54.2%   
6                           NaN  2006       2006             39.0%   
7                           NaN  2007       2007             27.0%   
8                           NaN  2008       2008             19.2%   
9                           NaN  2009       2009             16.2%   
10                          NaN  2010       2010             14.5%   
11                          NaN  2011       2011             13.2%   
12                          NaN  2012       2012             11.0%   
13                          NaN  2013       2013             10.3%   
14                          NaN  2014       2014              9.5%   
15                          NaN  2015       2015              9.3%   
16                          NaN  2016       2016             10.1%   
17                          NaN  2017       2017             11.8%   
18                          NaN  2018       2018              9.8%   
19                          NaN  2019       2019             10.3%   

   Format Value # (Billion) Format Value # (Million) Total Value # (Billion)  \
0                       NaN                  $942.5M                    1.1B   
1                       NaN                  $881.9M                     NaN   
2                       NaN                  $803.3M                     NaN   
3                       NaN                  $746.0M                     NaN   
4                       NaN                  $767.0M                     NaN   
5                       NaN                  $705.4M                    1.3B   
6                       NaN                  $619.7M                    1.6B   
7                       NaN                  $499.7M                    1.9B   
8                       NaN                  $368.4M                    1.9B   
9                       NaN                  $296.6M                    1.8B   
10                      NaN                  $253.0M                    1.7B   
11                      NaN                  $240.8M                    1.8B   
12                      NaN                

In [5]:
volume_df.columns

Index(['Year of Year Date', 'Format', 'Format (copy)', 'Metric',
       'Value (Actual)', 'Adjusted for Inflation Flag', 'Year', 'Year Date',
       '% of Total Volume', 'Format Value # (Billion)',
       'Format Value # (Million)', 'Total Value # (Billion)',
       'Total Value # (Million)', 'Total Value For Year',
       'Value (Actual) (copy)', 'Year (copy)'],
      dtype='object')

In [6]:
volume_df2 = volume_df.rename(columns={'Value (Actual)':'Units_Sold','% of Total Volume': 'Percent_Of_Total_Volume', 'Format Value # (Million)': 'Format_Value_Num_Million',
                                     'Total Value # (Billion)':'Total_Value_Num_Billion','Total Value # (Million)':'Total_Value_Num_Million',
                                     'Total Value For Year':'Total_Value_For_Year'})

In [7]:
volume_df2.columns

Index(['Year of Year Date', 'Format', 'Format (copy)', 'Metric', 'Units_Sold',
       'Adjusted for Inflation Flag', 'Year', 'Year Date',
       'Percent_Of_Total_Volume', 'Format Value # (Billion)',
       'Format_Value_Num_Million', 'Total_Value_Num_Billion',
       'Total_Value_Num_Million', 'Total_Value_For_Year',
       'Value (Actual) (copy)', 'Year (copy)'],
      dtype='object')

In [8]:
cols_to_keep= ['Format','Metric','Units_Sold','Year','Percent_Of_Total_Volume','Total_Value_Num_Million','Total_Value_Num_Billion','Total_Value_For_Year']

In [9]:
cols_to_keep

['Format',
 'Metric',
 'Units_Sold',
 'Year',
 'Percent_Of_Total_Volume',
 'Total_Value_Num_Million',
 'Total_Value_Num_Billion',
 'Total_Value_For_Year']

In [10]:
volume_df3=volume_df2[cols_to_keep]

In [11]:
volume_df3

Format Metric  Units_Sold  Year Percent_Of_Total_Volume  \
0                       CD  Units  942.500000  2000                   87.3%   
1                       CD  Units  881.900000  2001                   91.1%   
2                       CD  Units  803.300000  2002                   93.4%   
3                       CD  Units  746.000000  2003                   93.4%   
4                       CD  Units  767.000000  2004                   80.1%   
..                     ...    ...         ...   ...                     ...   
256  Ringtones & Ringbacks  Units   21.924866  2015                    1.7%   
257  Ringtones & Ringbacks  Units   22.620785  2016                    2.3%   
258  Ringtones & Ringbacks  Units   14.262870  2017                    1.9%   
259  Ringtones & Ringbacks  Units   10.026287  2018                    1.9%   
260  Ringtones & Ringbacks  Units    8.605770  2019                    1.9%   

    Total_Value_Num_Million Total_Value_Num_Billion Total_Value_For_Year  
0                       NaN                    1.1B             $1079.2B  
1                   $968.5M                     NaN              $968.5B  
2                   $859.7M                     NaN              $859.7B  
3                   $798.4M                     NaN              $798.4B  
4                   $958.1M                     NaN              $958.1B  
..                      ...                     ...                  ...  
256                     NaN                    1.3B             $1255.5B  
257                 $970.1M                     NaN              $970.1B  
258                 $742.6M                     NaN              $742.6B  
259                 $532.3M                     NaN              $532.3B  
260                 $453.3M                     NaN              $453.3B  

[261 rows x 8 columns]

### Notes:

* Renamed column titles to remove spaces and other signs such as % and #
* Removed columns that were unneeded, some were copies and some were just irrelevant. Others were missing data.
* Units_Sold will be the most relevant column in terms of quantifying sales. 
* Kept 'Total_Value_Num_Million','Total_Value_Num_Billion','Total_Value_or_Year' just for context, but will not be relevant when analyzing this data frame as the Revenue data frame will bring more context into sales in terms of Revenue. 

In [12]:
volume_df3['Percent_Of_Total_Volume']

0      87.3%
1      91.1%
2      93.4%
3      93.4%
4      80.1%
       ...  
256     1.7%
257     2.3%
258     1.9%
259     1.9%
260     1.9%
Name: Percent_Of_Total_Volume, Length: 261, dtype: object

In [13]:
volume_df3['Percent_Of_Total_Volume'] = volume_df3['Percent_Of_Total_Volume'].str.replace('%','')

In [14]:
volume_df3['Percent_Of_Total_Volume']

0      87.3
1      91.1
2      93.4
3      93.4
4      80.1
       ... 
256     1.7
257     2.3
258     1.9
259     1.9
260     1.9
Name: Percent_Of_Total_Volume, Length: 261, dtype: object

In [15]:
volume_df3

Format Metric  Units_Sold  Year Percent_Of_Total_Volume  \
0                       CD  Units  942.500000  2000                    87.3   
1                       CD  Units  881.900000  2001                    91.1   
2                       CD  Units  803.300000  2002                    93.4   
3                       CD  Units  746.000000  2003                    93.4   
4                       CD  Units  767.000000  2004                    80.1   
..                     ...    ...         ...   ...                     ...   
256  Ringtones & Ringbacks  Units   21.924866  2015                     1.7   
257  Ringtones & Ringbacks  Units   22.620785  2016                     2.3   
258  Ringtones & Ringbacks  Units   14.262870  2017                     1.9   
259  Ringtones & Ringbacks  Units   10.026287  2018                     1.9   
260  Ringtones & Ringbacks  Units    8.605770  2019                     1.9   

    Total_Value_Num_Million Total_Value_Num_Billion Total_Value_For_Year  
0                       NaN                    1.1B             $1079.2B  
1                   $968.5M                     NaN              $968.5B  
2                   $859.7M                     NaN              $859.7B  
3                   $798.4M                     NaN              $798.4B  
4                   $958.1M                     NaN              $958.1B  
..                      ...                     ...                  ...  
256                     NaN                    1.3B             $1255.5B  
257                 $970.1M                     NaN              $970.1B  
258                 $742.6M                     NaN              $742.6B  
259                 $532.3M                     NaN              $532.3B  
260                 $453.3M                     NaN              $453.3B  

[261 rows x 8 columns]

In [16]:
volume_df3["Percent_Of_Total_Volume"] = volume_df3.Percent_Of_Total_Volume.astype(float)

In [17]:
volume_df3['Percent_Of_Total_Volume']

0      87.3
1      91.1
2      93.4
3      93.4
4      80.1
       ... 
256     1.7
257     2.3
258     1.9
259     1.9
260     1.9
Name: Percent_Of_Total_Volume, Length: 261, dtype: float64

## Note:

* Removed '%' from Percent_Of_Total_Volume and turned contents into floats
* This number may be relevant for figuring out the format that is more prevalent within a year. 
    * So for example, in the year 2000 we can see that CD sales were 87.3% of total volume of sales in the year. So that format was the most popular. 

In [18]:
volume_df3

Format Metric  Units_Sold  Year  Percent_Of_Total_Volume  \
0                       CD  Units  942.500000  2000                     87.3   
1                       CD  Units  881.900000  2001                     91.1   
2                       CD  Units  803.300000  2002                     93.4   
3                       CD  Units  746.000000  2003                     93.4   
4                       CD  Units  767.000000  2004                     80.1   
..                     ...    ...         ...   ...                      ...   
256  Ringtones & Ringbacks  Units   21.924866  2015                      1.7   
257  Ringtones & Ringbacks  Units   22.620785  2016                      2.3   
258  Ringtones & Ringbacks  Units   14.262870  2017                      1.9   
259  Ringtones & Ringbacks  Units   10.026287  2018                      1.9   
260  Ringtones & Ringbacks  Units    8.605770  2019                      1.9   

    Total_Value_Num_Million Total_Value_Num_Billion Total_Value_For_Year  
0                       NaN                    1.1B             $1079.2B  
1                   $968.5M                     NaN              $968.5B  
2                   $859.7M                     NaN              $859.7B  
3                   $798.4M                     NaN              $798.4B  
4                   $958.1M                     NaN              $958.1B  
..                      ...                     ...                  ...  
256                     NaN                    1.3B             $1255.5B  
257                 $970.1M                     NaN              $970.1B  
258                 $742.6M                     NaN              $742.6B  
259                 $532.3M                     NaN              $532.3B  
260                 $453.3M                     NaN              $453.3B  

[261 rows x 8 columns]

In [19]:
volume_df3.to_csv('../../../data/Old_Data/Music_Sales_Volume_Clean.csv', index=False)